### ISSUE TITLE
Design Discussion

### ISSUE THREAD

**Issue by NickTikhonov** [education/classroom-assistant#11]
We should have a good idea of what the app is going to look at before getting started with React/PhotonKit. It would be good to have a sketch of the different views the app will show, and how teachers will be able to interact with the interface to get what they want done. Once we have a sketch, we can get feedback from the design team.

CC @johndbritton @tarebyte

**Comment by NickTikhonov** [comment]
Here are some sketches of what I think is a good starting point for the design. This design assumes that Classroom will provide a button on the assignment screen that will launch the desktop application. This design considers [Schneiderman's UI Heuristics (https://faculty.washington.edu/jtenenbg/courses/360/f04/sessions/schneidermanGoldenRules.html)

![new doc 1_1](https://cloud.githubusercontent.com/assets/6755555/16531391/26dd987e-3f81-11e6-883e-76c38847e582.jpg)

## View 2 - Assignment selection

* This view matches the assignment view in Classroom web.
* Allows teachers to confirm that the correct assignment is being archived
* Teachers can select all submissions with one click, or individual submissions by viewing the list of all available
* Keeping submission selection within the desktop app helps improve the workflow if the user makes an incorrect selection (ability to go back and change selection vs. quitting the app and going back to the browser)

## View 3 - Summary / Archive Destination

*  User is presented with feedback on the selections made on the previous screen "4/10 selected"
*  User is given the option of choosing a custom destination for the archive (default configured in settings)

## View 4 - Archiving Progress

* This view shows job progress
* A list of selected items with their individual progress
* When downloaded, the items flash a different color and disappear from the list
* Quitting at this point must present a confirmation dialog - the program must avoid leaving the directory in an inconsistent state, archived assignments will be removed if the app is closed. 

![new doc 1_2](https://cloud.githubusercontent.com/assets/6755555/16531853/417845ce-3f83-11e6-89d8-f2343083ce70.jpg)

## View 5 - Archiving Complete

* This is a continuation of View 4
* When all items disappear, the list gets replaced with a confirmation pane
* The user can choose to view the files with the system viewer or return to the home screen

## View 0 - Home Screen

* Show when the app is launched by the user (e.g. icon clicked)
* Button for first-time setup if app is opened for the first time
* Button that opens the classroom website in a browser window otherwise

## View 0.1 - Setup

* Settings view

CC @johndbritton @tarebyte

**Comment by tarebyte** [comment]
@NickTikhonov this looks good to me, it seems really straight forward and pretty easy to use.

**Comment by NickTikhonov** [comment]
Notes on the design from @johndbritton:

* Most common scenario is the teacher cloning all of the submissions - the checkboxed on view 2 should be checked by default. 

* Prior to downloading submission repos, a new directory should be created at the destination path with the format `<assignment name>-<current datetime>` to ensure there are no clashes from multiple downloads of the same assignment. 

* Views 4 and 5 are merged into one, with the behavior slightly changed - instead of items disappearing once downloaded, they stay - with the percentage on the far-right being replaced with a button to show that folder in the system FS viewer. That view effectively doubles as a submission browser once cloning is complete.
